analysis/bulk-deconvolution/exploratory-notebooks/quantiseq-tumor-genes.Rmd

```diff
@@ -0,0 +1,264 @@
+      }
+  ) |> 
+  # Combine into a single data frame
+  purrr::reduce(dplyr::left_join, by = "gene_symbol") |>
+  # Filter to signature genes only and add indicator column for tumor genes
+  dplyr::filter(gene_symbol %in% signature_genes) |>
+  dplyr::mutate(tumor_gene = gene_symbol %in% tumor_genes) |>
+  # Long data frame
+  tidyr::pivot_longer(
+    starts_with("SCPCS"), 
+    names_to = "sample_id", 
+    values_to = "tpm"
+  )
+
+head(tpm_df)
+```
+
+And confirm once more that we have all the genes:
+
+```{r}
+all(signature_genes %in% tpm_df$gene_symbol)
+```
+
+## Expression distributions
+
+First, we'll visualize distributions of TPM for tumor and non-tumor genes.
+
+
+```{r fig.height=12, fig.width=16}
+ggplot(tpm_df) + 
+  aes(x = tumor_gene, y = tpm, color = tumor_gene) + 
+  geom_jitter(width = 0.1, size = 0.25) + 
+  facet_wrap(~sample_id, scales = "free_y") + 
+  theme(
+    axis.text = element_text(size = rel(0.8)), 
+    axis.title = element_text(size = rel(0.8)), 
+    legend.position = "none"
+  )
+```
+
+Looking over a great many jitter plots, there does not seem to be a universal trend of abberantly high expression among tumor genes.
+
+
+## Permutation tests
+
+Next, we'll perform two flavors of permutation tests to test the hypothesis that tumor genes have higher expression compared to non-tumor genes.
+For each sample, we'll perform a parametric test using the t-statistic and nonparametric test using the u-statistic (aka, wilcox), doing `r params$permutation_reps` permutations each.
+
+We'll do these tests as one-sided to test if the tumor genes have higher expression.
+Code-wise, since the reference tumor genes level here is `FALSE`, we'll specify `alternative = "less"` for the one-sided test that non-tumor genes have lower expression than do tumor genes.
+
+```{r}
+samples <- unique(tpm_df$sample_id)
+
+perm_df <- samples |>
+  purrr::map(
+    \(x) {
+      sample_tpm <- dplyr::filter(tpm_df, sample_id == x)
+      
+      # Create permutations by shuffling tpm values
+      perms <- modelr::permute(sample_tpm, params$permutation_reps, tpm)
+      
+      # nonparametric
+      models <- purrr::map(perms$perm, \(perm_data) wilcox.test(tpm ~ tumor_gene, alternative = "less", data = perm_data))
+      glanced <- purrr::map_df(models, broom::glance, .id = "id")
+      mod <- wilcox.test(tpm ~ tumor_gene, alternative = "less", data = sample_tpm)
+      wilcox_p <- mean(glanced$statistic > broom::glance(mod)$statistic)
+      
+      # parametric
+      models <- purrr::map(perms$perm, \(perm_data) t.test(tpm ~ tumor_gene, alternative = "less", data = perm_data))
+      glanced <- purrr::map_df(models, broom::glance, .id = "id")
+      mod <- t.test(tpm ~ tumor_gene, alternative = "less", data = sample_tpm)
+      ttest_p <- mean(glanced$statistic > broom::glance(mod)$statistic)
+      
+      data.frame(
+        sample_id = x, 
+        nonparametric = wilcox_p, 
+        parametric = ttest_p
+      )
+    }
```

Comment:
Expanding on my earlier comment, just for curiosity, what would happen if you just did this (just the straight tests, no permutations).
```suggestion
  purrr::map(
    \(x) {
      sample_tpm <- dplyr::filter(tpm_df, sample_id == x)
      
      wilcox_p <- wilcox.test(tpm ~ tumor_gene, alternative = "less", data = sample_tpm)$p.value
      ttest_p <- t.test(tpm ~ tumor_gene, alternative = "less", data = sample_tpm)$p.value
      
      data.frame(
        sample_id = x, 
        nonparametric = wilcox_p, 
        parametric = ttest_p
      )
    }
```

Reply:
Hmm...well.... 
![Screenshot 2025-01-08 at 1 57 33 PM](https://github.com/user-attachments/assets/539996ca-8b2f-4aa9-8d5c-62b0cd94a324)


I made sure the directionality was right, too:

Reply:
Okay, so that makes me a bit concerned that the permutations are not quite right. To be honest, I was not sure I followed the logic of your permutation code.

Looking back at it now, I think you actually have the directionality of the statistic comparisons backward. You should be looking for the observed statistic values to be _greater_ than the permutations.

Reply:
> You should be looking for the observed statistic values to be greater than the permutations.

Ahhhhaaa that is the bug, glad we caught this, as I'm more inclined to think we should turn tumor mode on! I would tend to prefer to pick one setting and use it throughout all samples, but do you want to convince me otherwise?